### PR TITLE
Modify rule S6177: improve rule description

### PR DESCRIPTION
**Pull request by mostafa-mohammed-sonarsource**
## Review

A dedicated reviewer checked the rule description successfully for:

- [x] logical errors and incorrect information
- [x] information gaps and missing content
- [x] text style and tone
- [ ] PR summary and labels follow [the guidelines](https://github.com/SonarSource/rspec/#to-modify-an-existing-rule)

### CODE REVIEW

File: rules/S6177/cfamily/rule.adoc [hunk 1]
```diff
@@ -17,18 +17,24 @@ std::string_view toString(rgbaColorChannel channel) {
...
 
+This rule reports scopes that use a particular ``++enum class++`` extensively and can benefit from ``++using enum++`` declaration. For example, it reports most ``++switch++`` statements applied to an ``++enum++`` value.
 
-This rule reports scopes that use a particular ``++class enum++`` extensively and could benefit from ``++using enum++`` declaration. For example, it reports most ``++switch++`` statements applied to an ``++enum++`` value.
+=== Exceptions
 
+The rule does not apply if adding the `using enum` clause would create a name collision or reduce readability by shadowing a name.
 
-=== Noncompliant code example
+== How to fix it
```
Comment (alejandro-alvarez-sonarsource) [inline rules/S6177/cfamily/rule.adoc:32]: I'd add a short how here, such as 

>  Add `using enum` inside the switch statement and replace the qualified identifiers on the `case` labels with the corresponding enumerator names.

Or something of the sort.

File: rules/S6177/cfamily/rule.adoc [hunk 2]
```diff
@@ -17,18 +17,27 @@ std::string_view toString(rgbaColorChannel channel) {
...
 
-[source,cpp]
+Add `using enum` inside the relevant scope and replace teh qualified identifiers with the corresponding enumerator names.
```
Comment (alejandro-alvarez-sonarsource) [inline rules/S6177/cfamily/rule.adoc:34]:
```suggestion
Add `using enum` inside the relevant scope and replace the qualified identifiers with the corresponding enumerator names.
```

**Review by alejandro-alvarez-sonarsource — APPROVED** [review]
Just a small typo.

**Review by alejandro-alvarez-sonarsource — APPROVED** [review]
Just a small typo.

**Comment by alejandro-alvarez-sonarsource** [comment]
Also, before merging, rename the PR to [follow the guidelines](https://github.com/SonarSource/rspec/#to-modify-an-existing-rule). (i.e. Modify rule S6177 ...)

**Comment by tomasz-kaminski-sonarsource** [comment]
I would like to also add a non-switch example that is mentioned in the ticket. We see a lot of comments in SC feedback when this issue is rejected with a "is not a switch" message.

**Pull request merged**